build.gradle

```diff
@@ -147,3 +148,28 @@ jacocoTestReport {
         xml.required = true
     }
 }
+
+diffCoverageReport {
+    diffSource { // Only one source may be specified
+        // This requires files to have been committed
+        // https://github.com/form-com/diff-coverage-gradle/issues/73
+        git.compareWith 'refs/remotes/origin/main'
+
+        // To compare uncommitted files, uncomment the below:
+        // file = 'build/tmp/git.diff'
+        // And then run:
+        //   git diff refs/remotes/origin/main > build/tmp/git.diff
+        //   ./gradlew diffCoverage
+    }
+
+    // View report at build/reports/jacoco/diffCoverage/html/index.html
+    reports {
+        html = true
+    }
+
+    violationRules {
+        minBranches = 0.9
+        minLines = 0.9
+        minInstructions = 0.9
+    }
+}
```

Comment:
Should we add codecoverage dependency on gradle check here?
```
check.dependsOn jacocoTestCoverageVerification

Reply:
Given the check has to occur to generate the coverage before this happens, that would create a circular dependency. :-)

We could add this check to the gradle check workflow if we wanted, but... I don't think we want that. There are some cases where it's impossible to test, and diff coverage can swing wildly.  If someone submitted a PR just to catch an exception, it might be impossible to add coverage and the check would always fail. 

This functionality duplicates what CodeCov is already doing on PRs, so we will have that information avilable when reviewing.  Coverage is a guideline and not a hard and fast rule.

Reply:
Also, I believe the requirement to have a gradle check (or at least test) run first is also [a bug]() so I'm not sure we will need dependencies going the other way.   Forcing this workflow to depend on gradle check would make it take longer if you've already been working to get passing tests.

Reply:
Sure. That makes sense. I feel like we should have a strict dependency on code coverage the way AD is doing [here](https://github.com/opensearch-project/anomaly-detection/blob/main/build.gradle#L615) and [here](https://github.com/opensearch-project/anomaly-detection/blob/main/build.gradle#L644). This will help us to improve the coverage of the repo. Although, we have have code coverage I am not sure if we care about coverage a lot. This will help us to overcome that. We can definitely cover this in a new issue

Reply:
@owaiskazi19 sure, open a discuss issue with a link to those AD suggestions and we can easily update our config later.

Reply:
Added https://github.com/opensearch-project/opensearch-sdk-java/issues/250 to discuss. Updated the current test to fail based on AD thresholds.

Reply:
Thanks @dbwiddis for opening up the issue!